run/cifs_archive/copy-music.sh

```diff
@@ -65,11 +69,27 @@ do
   fi
 done <<< "$(cd "$SRC"; find * -type f)"
 
+# Delete files from the local partition(DST) files that do not exist in the music archive(SRC).
+while read file_name
+do
+  if [ ! -e "$SRC/$file_name" ]
+  then
+    if rm "$DST/$file_name"
+    then
+      NUM_FILES_DELETED=$((NUM_FILES_DELETED + 1))
+    else
+      log "Couldn't delete $DST/$file_name"
+      NUM_FILES_DELETE_ERROR=$((NUM_FILES_DELETE_ERROR + 1))
+    fi
+  fi
+done <<< "$(cd "$DST"; find * -type f)"
+
+# Stop the connection monitor.
```

Comment:
would it be a good idea to do this before copying the new files?

Reply:
I'm have been going back and forth on this but I think I'm going to agree to do the deletes first. 

The main use case I can see for someone wanting the music to be deleted is that they are rolling through a large collection of music/audio. In that case doing the delete first frees space for the new music/audio that the user is going to add.

The big downside is that if the archive disconnects after the delete but before the copy of new files finishes the user may not have much audio on the drive at all. I think avoiding the failure to copy is the more "right thing".

Update coming soon.